lib/src/package_meta.dart

```diff
@@ -274,61 +269,52 @@ abstract class PubPackageMeta extends PackageMeta {
     return isSdk ? 'Dart' : null;
   }
 
-  String _resolvedDir;
-
   @override
-  String get resolvedDir {
-    _resolvedDir ??= dir.resolveSymbolicLinksSync().path;
-    return _resolvedDir;
-  }
+  late final String resolvedDir = dir.resolveSymbolicLinksSync().path;
 }
 
 class _FilePackageMeta extends PubPackageMeta {
-  File _readme;
-  File _license;
-  File _changelog;
-  Map<dynamic, dynamic> _pubspec;
+  File? _readme;
+  File? _license;
+  File? _changelog;
 
-  _FilePackageMeta(Folder dir, ResourceProvider resourceProvider)
-      : super(dir, resourceProvider) {
+  late final Map<dynamic, dynamic> _pubspec = () {
     var pubspec = dir.getChildAssumingFile('pubspec.yaml');
     assert(pubspec.exists);
```

Comment:
Ha, interesting. This wouldn't need to use an anonymous closure, except for the assert statement... just interesting.

Reply:
Yes.  I'm a little bit torn on leaving the assert or not, but in general these yaml parsing file interaction classes tend to be really hard to debug so I've historically been liberal with the asserts -- based on the fail early principle.

Reply:
yeah I definitely love asserts for the reason you described.